cxx4opencl/diff2openclc.txt

```diff
@@ -129,6 +129,18 @@ label:  // invalid: jumping forward over declaration of n
    // ...
 ----------
 
+===== Ternary Selection Operator
+
+The ternary selection operator (`?:`) inherits its behaviour from both
+{cpp} and OpenCL C. It operates on three expressions
+`(exp1 ? exp2 : exp3)`. If all three expressions are scalar values,
+the {cpp}17 rules for ternary operator are followed. If the result is
+a vector value, then this is equivalent to calling
+`select(exp3, exp2, exp1)` as described in `OpenCL C v2.0 s6.13.6`.
+The rules from OpenCL C impose limitation that `exp1` cannot be a
+vector of float values. However, `exp1` can be evaluated to a scalar
```

Comment:
Btw is a vector of float allowed? Is this something we should mention explicitly?

Reply:
No, vector of floats is not allowed because we reuse logic from OpenCL C. I can reword this sentence to be more explicit about C++ for OpenCL:
```suggestion
vector of float values so it is also disallowed in C++ for OpenCL. However, `exp1` can be evaluated to a scalar
```

Reply:
Maybe we  can leave out what is identical to OpenCL and only document the differences since this section is dedicated to differences in behavior?

Reply:
While this discusses some improvements, there is no incorrect wording in this PR, and since we are close to the release date I would suggest to merge it as it and any improvements can be made separately.

Reply:
I think it is good to be explicit here about what is identical and what differs between languages because ternary operator inherits its behavior from both C++ and OpenCL C and differs from both of them. I feel that direct comparison with the two component languages makes it easier to explain how operator acts.